adapters/http-vertx-base/src/main/java/org/eclipse/hono/adapter/http/AbstractVertxBasedHttpProtocolAdapter.java

```diff
@@ -526,4 +538,15 @@ private void doUploadMessage(final RoutingContext ctx, final String tenant, fina
             });
         }
     }
+
+    private static Integer getQoSLevel(final String qosValue) {
+        try {
+            if (qosValue == null)
+                return null;
+            else
+                return Integer.parseInt(qosValue);
+        } catch (NumberFormatException e) {
+            return HEADER_QOS_INVALID;
```

Comment:
introducing the `INVALID` is a good idea, you should also use it for all values other than the allowed one (`AT_LEAST_ONCE`). You can then simplify the check above ...

Reply:
@sophokles73: please note that if the qos header is not set (like in all the existing test cases before this patch), then `ctx.request().getHeader(QOS_HEADER)` will be `null`. imo this is not an invalid header and is why am not using `INVALID` when the header is not set. however, when the header is set but cannot be parsed, then its `INVALID`.

what do you think?

Reply:
I understand, my point is that you can use `INVALID` for both a value that is not a number at all as well as for any number != 1.

Reply:
ok, i see. thanks

Reply:
how about this approach (before i update the PR)?:
```java
try {
       if (qosValue == null) {
          return null;
       }
       else {
          return Integer.parseInt(qosValue) != AT_LEAST_ONCE ? HEADER_QOS_INVALID : AT_LEAST_ONCE;
       }
} catch (NumberFormatException e) {
      return HEADER_QOS_INVALID;
}
```

Reply:
that should do the trick. but please format like this
```
try {
       if (qosValue == null) {
          return null;
       } else {
          return Integer.parseInt(qosValue) != AT_LEAST_ONCE ? HEADER_QOS_INVALID : AT_LEAST_ONCE;
       }
} catch (NumberFormatException e) {
      return HEADER_QOS_INVALID;
}
```